src/core/expr/cumsum.cc

```diff
@@ -0,0 +1,110 @@
+#include "column/virtual.h"
+#include "documentation.h"
+#include "expr/fexpr_func.h"
+#include "expr/eval_context.h"
+#include "expr/workframe.h"
+#include "python/xargs.h"
+#include "stype.h"
+namespace dt {
+namespace expr {
+
+static auto sum {0};
```

Comment:
Implementing `dt.cumsum()` based on the static variable will not work. Try to call this function twice on the same data or simply look at the random row and this is what you will see:

```python
>>> from datatable import dt
>>> DT = dt.Frame(range(5))
>>> DT[:, dt.cumsum(dt.f[0])]
   |    C0
   | int32
-- + -----
 0 |     0
 1 |     1
 2 |     3
 3 |     6
 4 |    10
[5 rows x 1 column]
>>> DT[:, dt.cumsum(dt.f[0])]
   |    C0
   | int32
-- + -----
 0 |    10
 1 |    11
 2 |    13
 3 |    16
 4 |    20
[5 rows x 1 column]
```

or

```python
>>> from datatable import dt
>>> DT = dt.Frame(range(5))
>>> DT[1, dt.cumsum(dt.f[0])] # Note, the result should be `1`
   |    C0
   | int32
-- + -----
 0 |     2
[1 row x 1 column]
>>> 
```

Reply:
This is because the result is accumulated in the static variable between the calls. Also, there is no loop to calculate `cumsum`, so accessing the random element will not actually do the supposed work. 

If you prefer, I can adjust your code to make it work in the way it is designed now (through the virtual columns) and we will start improving the code from there.

Reply:
@oleksiyskononenko yes, let's do that. it will help me understand more